core/commands/partials/terraform_configurator.rb

```diff
@@ -23,11 +23,14 @@ def initialize(config, env, logger)
     @machine_configurator = MachineConfigurator.new(@ui)
     @attempts = @env.attempts&.to_i || 5
     @recreate_nodes = @env.recreate
-    @network_settings = if File.exist?(config.network_settings_file)
-                          NetworkSettings.from_file(config.network_settings_file)
-                        else
-                          NetworkSettings.new
-                        end
+    if File.exist?(config.network_settings_file)
```

Comment:
The check is already present in the `from_file` method. You do not need to check here too.